test/graphql/resolvers/volunteer_resolver_test.rb

```diff
@@ -0,0 +1,163 @@
+require_relative '../../test_helper'
+
+SingleCov.covered!
+
+describe VolunteerResolver do
+  let(:user) { users(:admin) }
+  let(:user2) { users(:a) }
+  let(:user3) { users(:b) }
+  let(:user4) { users(:cat) }
+  let(:user5) { users(:minimum_volunteer) }
+  let(:office) { offices(:remote) }
+  let(:office2) { offices(:madison) }
+
+  describe '.all' do
+    before do
+      Signup.delete_all
+      IndividualEvent.delete_all
+      user.update(office: office, role: Role.admin)
+    end
+
+    let(:event1) { events(:minimum) }
+    let(:event2) { events(:simple_event) }
+    let(:event3) { events(:late_event) }
+
+    it 'limits with time bounds' do
+      event1.update(starts_at: 2.weeks.ago, ends_at: (2.weeks.ago + 120.minutes))
+      event2.update(starts_at: 4.days.ago, ends_at: (4.days.ago + 20.minutes))
+      event3.update(starts_at: 3.days.ago, ends_at: (3.days.ago + 30.minutes))
+
+      Signup.delete_all
+      Signup.create!(user: user, event: event1)
+      Signup.create!(user: user, event: event2)
+      Signup.create!(user: user2, event: event3)
+
+      args = { after: 1.week.ago.to_i, before: 2.days.ago.to_i, sort_by: 'HOURS_DESC' }
+      results = VolunteerResolver.all(nil, args, nil).to_a
+
+      results.must_equal [user2, user]
+    end
+
+    it 'filters by all offices' do
+      args = { office_id: 'all' }
+
+      Signup.create!(event: event1, user: user)
+      Signup.create!(event: event2, user: user2)
+      Signup.create!(event: event1, user: user3)
+      Signup.create!(event: event2, user: user4)
+      Signup.create!(event: event1, user: user5)
+
+      results = Set.new(VolunteerResolver.all(nil, args, nil).to_a)
+      users = Set.new(User.all)
+
+      results.must_equal users
+    end
+
+    it 'filters by current office' do
+      args = { office_id: 'current' }
+      context = { current_user: user }
+
+      Signup.create!(event: event1, user: user)
+      results = VolunteerResolver.all(nil, args, context).to_a
+
+      results.must_equal [user]
+    end
+
+    it 'filters by office_id' do
+      args = { office_id: office.id }
+
+      Signup.create!(event: event1, user: user)
+      results = VolunteerResolver.all(nil, args, nil).to_a
+
+      results.must_equal [user]
+    end
+
+    it 'limits to count' do
+      args = { count: 0 }
```

Comment:
This test could generate a false positive if there was something broken. Would be better to test both `0` and `n` case.